src/lock_file/satisfiability.rs

```diff
@@ -20,8 +20,8 @@ use pixi_manifest::FeaturesExt;
 use pixi_record::{LockedGitUrl, ParseLockFileError, PixiRecord, SourceMismatchError};
 use pixi_spec::{PixiSpec, SourceSpec, SpecConversionError};
 use pixi_uv_conversions::{
-    as_uv_req, as_uv_specifiers, as_uv_version, into_pixi_reference, to_normalize,
```

Comment:
If an environment is solved with `no-build = false` and later that changes to true, we should resolve if we depend on sdists. Assuming we can just look for `tar.gz` that shouldn't be to hard right? 
We can skip the other way around as that doesn't invalidate the manifest with the lockfile and a `pixi update` should fix those issues.